NetworkRequest/NetworkRequest/ViewController.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  ViewController.swift
+//  NetworkRequest
+//
+//  Created by Ivan on 02.08.2023.
+//
+
+import UIKit
+
+class ViewController: UIViewController, UISearchBarDelegate {
+
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    private var repositories: [Repository] = []
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchBar.delegate = self
+        tableView.dataSource = self
+        tableView.delegate = self
+    }
+    
+    @objc func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let searchText = searchBar.text else { return }
+        searchRepositories(withQuery: searchText)
+    }
+    
+    func searchRepositories(withQuery query: String) {
```

Comment:
This func could be private since it is used only in the scope of this class.
Actually it is even would be better if you'll create a separate class for it and move it there, for example `GitHubApiAdapter` or `GitHubService`, then it shouldn't be private